types/react-cache/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for react-cache 2.0
+// Project: https://github.com/facebook/react/tree/master/packages/react-cache
+// Definitions by: Spencer Miskoviak <https://github.com/skovy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Thenable<Value> {
+  then(resolve: (value: Value) => any, reject: (reason: any) => any): any;
+}
```

Comment:
In TS there's already a global declaration for Thenable which is more type-safe than this (it's `PromiseLike<T>`)